odoo/addons/base/ir/ir_cron.py

```diff
@@ -209,6 +212,18 @@ def _process_jobs(cls, db_name):
                               ORDER BY priority""")
                 jobs = cr.dictfetchall()
 
+            if changes:
+                if not jobs:
+                    raise BadModuleState()
+                # nextcall is never updated if the cron is not executed,
+                # it is used as a sentinel value to check whether cron jobs
+                # have been locked for a long time (stuck)
+                oldest = min([parse(job['nextcall']) for job in jobs])
+                if oldest + MAX_FAIL_TIME < datetime.now():
```

Comment:
I find `datetime.now() - oldest > MAX_FAIL_TIME` more readable, but that's a personal taste ;-)